### PR TITLE
CMO-152: New stateful component

### PR DESCRIPTION
**Pull request by edgibbs**
Some overall notes:

* This is a draft of adding the inline validation for non-valid dates
* The tests are incomplete
* There are probably a few extra things that can still be refactored
* The DateField component can probably be deleted as well since this is the only usage of it
* I need a more recent rebase

### Jira Story

- [Name of Story INT-NNN](https://osi-cwds.atlassian.net/browse/INT-NNN)

## Tests
- [ ] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/common/DateFieldStateful.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      this.props.onChange(null)
+      return
+    }
+
+    this.props.onChange(dateFormatToYYYYMMDD(date))
+  }
+
+  handleOnBlur() {
+    if (!_.isEmpty(this.state.shadowValue) && Number.isNaN(Date.parse(this.state.shadowValue))) {
+      this.setState({errors: ['Please enter a valid date']})
```
Comment (pokharelmanish) [inline app/javascript/common/DateFieldStateful.jsx]: Do we have a test for the this ('Please enter a valid date')?

File: app/javascript/common/DateFieldStateful.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    } else {
+      this.setState({errors: []})
+    }
+
+    this.props.onBlur()
+  }
+
+  handleOnKeyUp(event) {
```
Comment (pokharelmanish) [inline app/javascript/common/DateFieldStateful.jsx]: test?

**Comment by pokharelmanish** [comment]
I entered `123456` in the date of birth field and when I blur out I don't see the validation error. 
Screen Shot 2019-11-05 at 11 06 45 AM

No validation error
Screen Shot 2019-11-05 at 11 06 51 AM

**Comment by TylerOSI** [comment]
@pokharelmanish did you click out of the field or blur another way?

**Comment by pokharelmanish** [comment]
@TylerOSI 
1. I clicked out of the field (didn't work)
2. I tabbed out of the field (didn't work)

**Comment by TylerOSI** [comment]
good suggestion. im going to merge this though, so that I can move onto more things.

**Pull request merged**